### PR TITLE
Property filter URL disconnect version 2

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Simpler version of https://github.com/PostHog/posthog/pull/7985
- This is the last feature in insights that modifies the URL independent of `insightLogic`.
- This unlocks fixing
  - The issue where we accidentally try to override filters on insights to {}, causing `dashboard.js` cypress tests to fail. It may already be fixed.
  - https://github.com/PostHog/posthog/issues/8000

## How did you test this code?

I manually tested that these still work, reloading the URL and going back/forward when applicable:

- [x] Trend, funnel, etc step property filters
- [x] Insight Paths exclusion filters
- [x] Global filters in insights
- [x] Events table property filters
- [x] Filters when adding/editing a cohorts
- [x] Persons page
- [x] Action editing
- [x] Experiments
- [x] Feature Flags
- [x] Session recordings
- [x] Test account filters in settings

### CODE REVIEW

File: frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts [hunk 1]
```diff
@@ -63,56 +57,21 @@ export const propertyFilterLogic = kea<propertyFilterLogicType>({
...
         },
     }),
 
     selectors: {
         filledFilters: [(s) => [s.filters], (filters) => filters.filter(isValidPropertyFilter)],
+        filtersWithNew: [
+            (s) => [s.filters],
+            (filters) => {
+                if (filters.length === 0 || isValidPropertyFilter(filters[filters.length - 1])) {
+                    return [...filters, {}]
```
Comment (pauldambra) [inline frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts:70]: adding in the empty object is a bit confusing (the name filtersWithNew threw me...)

but it's not new here, just made more visible, so no reason to block

Comment (pauldambra) [reply frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts:70]: Might be nicer to expose `filters.length === 0 || isValidPropertyFilter(filters[filters.length - 1])` and let the UI show the new filter when it is true

**Review by pauldambra — APPROVED** [review]

**Review by Twixes — APPROVED** [review]
I'm of the same opinion as Paul

File: frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts (hunk 1 above)
Comment (mariusandra) [reply frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts:70]: You mean like exposing `filtersWithNew`? :)

Given the two green checkmarks, I'll merge it in as in.

**Pull request merged**